zerver/views/streams.py

```diff
@@ -410,6 +409,9 @@ def stream_exists_backend(request, user_profile, stream_name, autosubscribe):
     stream = get_stream(stream_name, user_profile.realm)
     result = {"exists": bool(stream)}
     if stream is not None:
+        if stream.invite_only and stream.num_subscribers() == 0:
+            do_deactivate_stream(stream)
+            return json_response(data=result, status=404)
```

Comment:
This logic seems wrong -- stream_exists_backend shouldn't mutate state.

I think better logic would be to deactivate any invite-only stream when the last user leaves, rather than trying to catch it when someone tries to subscribe again later.